### ISSUE TITLE
Allow selective enforcement of require-unicode-regexp for literals and/or RegExp constructor

### ISSUE THREAD

**Issue by outoftime** [eslint/eslint#10959]
**What rule do you want to change?**

`require-unicode-regexp`

**Does this change cause the rule to produce more or fewer warnings?**

Fewer, if new options are used.

**How will the change be implemented? (New option, new default behavior, etc.)?**

New option

**Please provide some example code that this change will affect:**

```js
const a = /aaa/
const b = /bbb/gi
const c = new RegExp("ccc")
const d = new RegExp("ddd", "gi")
```

**What does the rule currently do for this code?**

Warns in all cases

**What will the rule do after it's changed?**

Depending on options used, only warn for the first two lines, or only warn for the second two lines.

The main motivation here is that Babel is able to transform Unicode regular expression literals for compatibility with older browsers, but there is, as far as I can tell, no existing polyfill to do the same for the `RegExp` constructor. So, for people who are targeting browsers that don’t natively support the `u` flag, it may be sensible to enable this rule only for literals.

**Comment by platinumazure** [comment]
This seems reasonable to me. :+1:
